### ISSUE TITLE
PG rating showing as R

### ISSUE THREAD

**Issue by meesterexx** [beatmasterRS/skin.arctic.zephyr.mod#146]
PG rated movies as showing as R. I noticed there was a textures cleanup 3 days ago, could this be related? I tried to decompile textures.xbt, but couldn't get the kodi texture tool to work (empty directory on decompile). The screenshot below is from TMDBHelper (v. 4.3.0) widget / Trending this Week, skin version 1.1.30. 
![rating](https://user-images.githubusercontent.com/13154733/112210102-a913ef80-8bf0-11eb-9efb-5b067f49ee57.png)

**Comment by beatmasterRS** [comment]
Thanks for pointing this out. Could you please enable debbuging in skin settings:

![screenshot00659](https://user-images.githubusercontent.com/3855273/112211586-5cdea480-8c1c-11eb-901d-ad5088002625.png)

After that, please go to the widget, open the dialogvideo info and take a screenshot:

![screenshot00660](https://user-images.githubusercontent.com/3855273/112211760-7aac0980-8c1c-11eb-8547-9acd412f304a.png)

**Comment by meesterexx** [comment]
it's mapping to flags/mpaa/r.png
![rating2](https://user-images.githubusercontent.com/13154733/112212626-95b65380-8bf3-11eb-8376-b09ac7be5c27.png)


Only a completely unrelated note, I wasn't sure if this was a bug or working as designed, but forced views wasn't honoring my settings until manually edited the view orders in MyVideoNav.xml. I set movies, shows and episodes to Netflix (504) and seasons to Season Info (516) and locked views on all. It would revert the views to DoubleFlix (511) on sections of TMDBHelper I didn't explicitly set the view in (for example if I set the view in Movies/Popular, it honored it but if I went to another section like In Theaters, it didn't). Setting the views I wanted (504,516) in MyVideoNav order before (511) fixed it:
 <views>50,501,504,516,507,509,510,511,512,513,550,56,57,53,55,555,59,54,51,58,52,500,514,515,517,519,520</views>
 
 If you think this is a bug, I am happy to open another issue so as not to confuse this issue).

**Comment by beatmasterRS** [comment]
Thats strange, cause in the code, there should be match: `<value condition="String.Contains(ListItem.mpaa,Rated PG) | String.IsEqual(ListItem.Mpaa,PG)">flags/mpaa/pg.png</value>` I will create a testbranch so we can test this.

As for the 2nd issue: unfortunately the "forced views" function doesn't work properly depending on the language or special characters like "ß", so I won't work on that any further, but I can make the change to the view ID 's, that's not a problem. :)

**Comment by meesterexx** [comment]
It's also showing an R rating on some TV shows where there should be no mpaa rating at all. This is TMDBHelper/Trakt next episodes widget.

![rating3](https://user-images.githubusercontent.com/13154733/112321149-1756c180-8c86-11eb-855a-129b5c069ec1.png)

**Comment by beatmasterRS** [comment]
Clould you please test this branch? https://github.com/beatmasterRS/skin.arctic.zephyr.mod/archive/refs/heads/ratings-fix-testing.zip

![screenshot00672](https://user-images.githubusercontent.com/3855273/112326457-d32ee580-8cb4-11eb-9201-566e5d26e54a.png)

**Comment by beatmasterRS** [comment]
> It's also showing an R rating on some TV shows where there should be no mpaa rating at all. This is TMDBHelper/Trakt next episodes widget.

This i can not reproduce. For me, if there is no rating given on a movie, no rating falg is shown. If there is a "n" rating given, the nr flag is shown.

**Comment by meesterexx** [comment]
Before I do test, a further clue, this seems to be specific to TMDBHelper. I tried a 3rd party addon and the issue does not occur:

![rating4](https://user-images.githubusercontent.com/13154733/112327298-af0ade80-8c8b-11eb-8ea7-70381fc92c17.png)

![rating5](https://user-images.githubusercontent.com/13154733/112327329-b6ca8300-8c8b-11eb-8872-7e9b5acaa158.png)

**Comment by meesterexx** [comment]
Do you still want me to test or should I write this off as an issue with TMDBHelper?

**Comment by beatmasterRS** [comment]
I actually only read the infolabel ListItem.MPAA and so far everything works correctly. If this is a TMDB helper problem I don't know. Maybe a configuration problem, but I don't know that either.

**Comment by meesterexx** [comment]
No change with using the test zip,  and this time I started from a clean kodi 19. I only installed TMDBHelper 4.3.0 from Kodi repo (instead of Jurial's) and then installed your repo, but did not install skin from your repo, installed from the test zip instead. No configuration changes were made at all except enabling debug and the icons in the skin so I could see ratings and I did not change any settings in TMDBHelper.

The only other possible factor I could think of is I have this version installed as a portable version using the "-p" option so the data folder \kodi\portable_data rather than the standard "\AppData\Roaming\Kodi". Is there any possibility this could have an effect?

![rating6](https://user-images.githubusercontent.com/13154733/112335367-8e925280-8c92-11eb-88c3-36e8b79b6101.png)

**Comment by beatmasterRS** [comment]
I have no idea what is causing this behavior. Maybe we can test it. Best again with Raya:

https://github.com/beatmasterRS/skin.arctic.zephyr.mod/archive/refs/heads/variables-testing.zip

**Comment by beatmasterRS** [comment]
![screenshot00673](https://user-images.githubusercontent.com/3855273/112340527-e942a300-8cc0-11eb-9275-48b96120f252.png)

Do you have the same setting?

**Comment by meesterexx** [comment]
That didn't work. Doesn't map to a flag variable at all:
![vartest1](https://user-images.githubusercontent.com/13154733/112362804-361b7f00-8cab-11eb-8557-943beb3687d5.png)

To rule out any issue caused by the portable version, I uninstalled kodi 18 and installed kodi 19 and fresh installed TMDBHelper and tried the skin from the test zips and the repo.

So I went back to the master skin version and searched on flags/mpaa/r.png and the lines 274 and 322 had a bit that looked sketchy:
String.Contains(ListItem.mpaa,Rated) 
String.Contains(Container(9500).ListItem.mpaa,Rated) 

Notice there is no designation after Rating. I edited it out and Raya started mapping to /flags/mpaa/Rated PG.png instead of r.png. Of course it got no icon because Rated PG.png isn't there I believe. 

So I tried to edit 273 and 321 for PG to match the extra conditionals found in the R rating line to see if I could get it to map to /flags/mpaa/pg.png, but had no luck. 

BTW, on lines 273 and 321, this item has a capital M in the list item ListItem.Mpaa and none of the others do, they're all ListItem.mpaa. Not sure if it's case sensitive, but I corrected for that, and it still had no effect.

So to sum up, I think the bit it the r rating lines that I mentioned above is causing it to match PG to r.png because whatever is in the PG lines isn't quite right and it's mapping to Rated PG.png instead of pg.png.

**Comment by meesterexx** [comment]
That didn't work either. In fact the only rating that works is PG-13. R, PG and NR don't work with this one.

I appreciate the help on this. I guess we can drop it since I have a work around. I'm kind of surprised that it happens on a clean install, but you can't duplicate the issue. Hopefully no else runs into it.  Thanks again!

**Comment by meesterexx** [comment]
The merge with the fix for https://github.com/beatmasterRS/skin.arctic.zephyr.mod/pull/147/commits/22e5cd41dcd6995ccd12dfbcca45598d02e79810 fixed my issue once I deleted the TMDBHelper databases (*.db files) in 
Kodi\userdata\addon_data\plugin.video.themoviedb.helper\database.

All ratings working now.

**Comment by beatmasterRS** [comment]
Thanks for your feedback :)
